src/forms/FormControl.scss

```diff
@@ -104,166 +138,197 @@
-  box-shadow: var(--primer-borderInset-thin, inset 0 0 0 max(1px, 0.0625rem)) var(--color-border-default);
-  grid-template-rows: auto;
-  gap: var(--primer-controlStack-medium-gap-condensed, 8px);
-  align-items: center;
-  align-content: center;
-  padding-inline: var(--primer-control-medium-paddingInline-condensed, 8px);
-
-  .FormControl {
-    padding: unset;
-    background-color: transparent;
-    border: 0;
+  position: relative;
+  display: grid;
+
+  .FormControl--input-leadingVisualWrap {
+    position: absolute;
+    top: var(--base-size-8, 8px);
+    left: var(--base-size-8, 8px);
+    display: block;
+    width: var(--base-size-16, 16px);
+    height: var(--base-size-16, 16px);
+    color: var(--color-fg-muted);
+    pointer-events: none;
 
-    &:focus-visible,
-    &:focus {
-      outline: none;
-      box-shadow: none;
+    // octicon
+    .FormControl--input-leadingVisual {
+      display: block;
+      user-select: none;
     }
   }
 
-  &:focus-within {
-    @include focusBoxShadowInset(2px, var(--color-accent-fg));
-  }
+  // TODO: replace with new Button component
+  .FormControl--input-trailingAction {
+    position: absolute;
+    top: var(--base-size-4, 4px);
+    right: var(--base-size-4, 4px);
+    z-index: 4;
+    display: grid;
+    width: var(--primer-control-xsmall-size, 24px);
+    height: var(--primer-control-xsmall-size, 24px);
+    padding: 0;
+    color: var(--color-fg-muted);
+    cursor: pointer;
+    background: transparent;
+    border: 0;
+    border-radius: var(--primer-borderRadius-small);
+    transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+    transition-property: color, background-color, border-color;
+    align-items: center;
+    justify-content: center;
 
-  &.FormControl-fieldWrap--disabled {
-    color: var(--color-primer-fg-disabled);
-    background-color: var(--color-input-disabled-bg);
-  }
+    svg {
+      user-select: none;
+    }
 
-  &.FormControl-fieldWrap--invalid:not(:focus-within) {
-    box-shadow: var(--primer-borderInset-thin, inset 0 0 0 max(1px, 0.0625rem)) var(--color-danger-emphasis);
-  }
+    &[disabled] {
+      color: var(--color-primer-fg-disabled);
+      pointer-events: none;
+    }
 
-  // if leadingVisual is present
-  &.FormControl-fieldWrap--input-leadingVisual {
-    grid-template-columns: var(--base-size-16, 16px) minmax(0, auto);
+    &:hover {
+      background: var(--color-action-list-item-default-hover-bg);
+    }
 
-    .FormControl--input {
-      padding-inline-start: calc(var(--base-size-16, 16px) + var(--primer-control-medium-paddingInline-condensed, 8px));
+    &:active {
+      background: var(--color-action-list-item-default-active-bg);
     }
 
-    // if leadingVisual and trailingAction are present
-    &.FormControl-fieldWrap--input-trailingAction {
-      grid-template-columns: var(--base-size-16, 16px) minmax(0, auto) min-content;
+    // show vertical divider line between field and button
+    &::before {
```

Comment:
Should this divider line be an optional prop? 

<img width="285" alt="input field with divider line between button" src="https://user-images.githubusercontent.com/18661030/174495505-d20b37cf-f841-4347-9f19-2aa8bdf50417.png">